### ISSUE TITLE
Trouble installing on python 3.10

### ISSUE THREAD

**Issue by ivirshup** [theislab/ncem#159]
**Describe the bug**
I believe this package should be installable on python 3.10, since the pyproject.toml says:

```python
python = ">=3.8,<=3.10"
```

But pip in a fresh conda environment isn't letting me install:

```
mamba create -yn "ncem-env" python=3.10 
conda activate ncem-env
pip install ncem
````

```
ERROR: Ignored the following versions that require a different python version: 0.1.0 Requires-Python >=3.7,<3.10; 0.1.1 Requires-Python >=3.7,<3.9; 0.1.2 Requires-Python >=3.7,<3.9; 0.1.3 Requires-Python >=3.7,<3.9; 0.1.4 Requires-Python >=3.7,<3.9; 0.1.5 Requires-Python >=3.8,<=3.10
ERROR: Could not find a version that satisfies the requirement ncem (from versions: none)
ERROR: No matching distribution found for ncem
```

Maybe the versioning could be loosened up a bit?

**System [please complete the following information]:**

```
-----
session_info        1.0.0
-----
Python 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0]
Linux-5.15.0-82-generic-x86_64-with-glibc2.35
-----
Session information updated at 2023-09-25 14:56
```

**Comment by AnnaChristina** [comment]
do you need python 3.10 for a specific other dependency you want to use? I recently installed it on 3.9 and worked with ncem evrsion 0.1.5.

**Comment by ivirshup** [comment]
> do you need python 3.10 for a specific other dependency you want to use?

No. It's not urgent, just something that came up while setting up environments and seems like something that should work given the version specifications.

3.9 is working, may be a bigger issue soon since I believe numpy will stop supporting 3.9 in the near future.

**Comment by AnnaChristina** [comment]
there is another issue upcoming, namely that patsy is no longer maintained and the whole linear ncem model backend is dependent on patsy. had not yet have time to think about updated dependencies.

**Comment by AnnaChristina** [comment]
but nevertheless this 3.10 issue is def a bug, will look into it. Thanks!

**Comment by DomenicoSkyWalker89** [comment]
Hello @AnnaChristina,
I'm checking in to see if there are any updates. I'm asking because I'm using a dependency that requires Python version >= 3.10.2.

Thank you for your assistance.

Best regards,
Domenico

**Comment by giovp** [comment]
just pinging this as I was trying today but had to create new env with python3.9

**Comment by s-weissbach** [comment]
> there is another issue upcoming, namely that patsy is no longer maintained and the whole linear ncem model backend is dependent on patsy. had not yet have time to think about updated dependencies.

I checked the [github repo](https://github.com/pydata/patsy) of patsy and it seems to be maintained and should work with python version 3.6 - 3.13. Relaxing the python version requirements should be definitely possible.
